contracts/nois-gateway/src/contract.rs

```diff
@@ -176,8 +298,29 @@ fn receive_request_beacon(
     let router = RequestRouter::new();
     let RoutingReceipt {
         acknowledgement,
-        msgs,
-    } = router.route(deps, env, channel, after, origin)?;
+        mut msgs,
+    } = router.route(deps.branch(), env, channel_id.clone(), after, origin)?;
+
+    // Pay time
+    let payment_contract = PAYMENT_ADDRESSES.load(deps.storage, &channel_id)?;
+
+    let config = CONFIG.load(deps.storage)?;
+
+    let Coin { amount, denom } = config.price;
+    let amount_burn = amount.mul_ceil((50u128, 100)); // 50%
+    let amount_relayer = amount.mul_ceil((5u128, 100)); // 5%
+    let amount_rest = amount - amount_burn - amount_relayer; // 45%
+
```

Comment:
Would it be nice here to add some map on the state linking channel_id -> n_requested_beacons. And increment the value every time we receive a request on a specific channel_id. Could be a nice way to build some sales/premiums/leaderboards ... in the future ? Or do you think this is not the priority now

Reply:
I think what is nice here is gneralize

```
const PAYMENT_ADDRESSES: Map<&str, Addr>
```

from `Addr` to a `Customer` struct. This can then have the two fields `payment: Addr` and `requested_beacons: u64` for now. Having the extra struct makes it easy to extend later on.

Reply:
Customer struct is very nice!

Reply:
[a00cb2e](https://github.com/noislabs/nois-contracts/pull/196/commits/a00cb2ee6d65fd87e3e0d6dffb323073a91c5b85)

This also makes the code more compact. `requested_beacons` is not yet incremented but I'll add it.